docs/designs/concepts.md

```diff
@@ -0,0 +1,180 @@
+# Concepts Design Doc
+
+This document describes core concepts of ElasticDL.
+
+An ElasticDL model consists of two kinds of parameters:
+
+1. dense parameters, and
+1. embedding tables.
+
+A dense parameter is a dense tensor with a name.  An embedding table is a map from some ID to embedding vectors.  An embedding table also has a name.  Formalizing the concepts, we have
+
+- model = {dense parameter} + {embedding table}
+- dense parameter = tensor + name
+- embedding table = {id, tensor} + name
+
+where the curly braces denote zero or more.
+
+It is notable that in practice, we use the rule
+
+- embedding table = tensor + index + name
+- index = map[ID]address
```

Comment:
embedding table = tensor + index + name
So this `index` is not ID but address? 
What is the relationship with IndexedSlices? Is `indices` in `IndexedSlices` is this `index`?
```
message IndexedSlices {
  Tensor tensor = 1;
  repeated int64 indices = 2;
}
```

Reply:
Please refer to the comment [here](https://github.com/sql-machine-learning/elasticdl/pull/1715/files#r376240095).

After we reach a consensus, I will update the doc.